web/static/js/content_types/image_editor.js

```diff
@@ -56,37 +57,78 @@ class ImageEditor {
   }
 
   onSubmit (data) {
-    const editor = document.querySelector(`[data-thesis-content-id="${data.contentId}"`)
+    this.set(data.contentId, data)
+    this.closeTray()
+  }
+
+  getContent (ed) {
+    const type = ed.getAttribute('data-thesis-content-type')
+    if (type === 'image') {
+      return ed.querySelector('img').getAttribute('src')
+    } else {
+      return this.getUrlFromStyle(ed.style.backgroundImage)
+    }
+  }
+
+  uploadAndSet (data, page, callback) {
+    const imageUrl = this.determineImageUrl(data.content, page.origin)
+    if (!imageUrl) { callback(); return }
+
+    Net.post('/thesis/files/import', {image_url: imageUrl, image_name: data.name})
+    .then((response) => {
+      callback()
+      if (response.path.length > 0) this.set(data.name, {url: response.path}, data.meta)
+    })
+    .catch((error) => { callback(); window.alert(error) })
+  }
```

Comment:
this is the import uploader. we first need to determine how to treat the url (more about this later)

we post the url to the server (very similar code to the image tray uploader)